src/views/saphyr/SaphyrReception.vue

```diff
@@ -68,83 +60,18 @@ export default {
           throw exportSampleTubesResponse.errors
         }
 
-        let tractionTubesBarcodeList = exportSampleTubesResponse.deserialize.requests.map(r => r.barcode)
-        await this.handleTractionTubes(tractionTubesBarcodeList)
-
       } catch (error) {
         this.showAlert(error.message, 'danger')
       }
     },
-    async findTractionTubes() {
-      try {
-        await this.handleTractionTubes(this.getBarcodes())
-      } catch (error) {
-        this.showAlert(error.message, 'danger')
-      }
-    },
-    async handleTractionTubes (barcodeList) {
-      if (barcodeList === undefined || !barcodeList.length) {
-        throw Error(consts.MESSAGE_WARNING_NO_BARCODES)
-      }
-
-      let response = await this.getTractionTubesForBarcodes(barcodeList)
-
-      if (response.successful && !response.empty) {
-        let tubes = this.tractionTubes
-
-        // check that all barcodes are valid
-        this.checkBarcodes(tubes, barcodeList)
-
-        // check that all the barcodes are either samples (requests) or libraries
-        this.checkMaterialTypes(tubes, barcodeList)
-      } else {
-        throw Error(consts.MESSAGE_ERROR_GET_TRACTION_TUBES)
-      }
-    },
-    /**
-     * Check that the list of barcodes provided match barcodes we have available and alert those
-     * which are invalid
-     * @param {*} barcodesList A list of barcodes which a user has entered/scanned
-     */
-    checkBarcodes(tubes, barcodesList) {
-      if (tubes.length !== barcodesList.length) {
-        let validBarcodes = tubes.map((tube) => tube.barcode)
-        let invalidBarcodes = barcodesList.filter(
-          scannedBarcode => !validBarcodes.includes(scannedBarcode))
-        throw Error(consts.MESSAGE_ERROR_INVALID_BARCODES.concat(invalidBarcodes.join(', ')))
-      }
-    },
-    /**
-     * To be directed to the correct view, either samples or libraries, we need to ensure that all
-     * the tubes are of the same material type
-     * @param {*} tubes All the tubes returned from the query
-     * @param {*} barcodesList A list of barcodes which a user has entered/scanned
-     */
-    checkMaterialTypes(tubes, barcodesList) {
-      let isAllRequests = tubes.every(t => t.material.type == 'requests')
-      let isAllLibraries = tubes.every(t => t.material.type == 'libraries')
-
-      if (isAllRequests) {
-        this.$router.push({ name: 'SaphyrSamples', query: { barcode: barcodesList } })
-      } else if (isAllLibraries) {
-        this.$router.push({ name: 'SaphyrLibraries', query: { barcode: barcodesList } })
-      } else {
-        // We only want barcodes of the same type
-      throw Error(consts.MESSAGE_ERROR_SINGLE_TYPE)
-      }
-    },
     ...mapActions('traction/saphyr/tubes', [
```

Comment:
I've moved 'exportSampleExtractionTubesIntoTraction' into requests vuex subfolder, if you want to do the same? It will help us move towards removing tubes vuex, and the request goes to /v1/requests too

Reply:
Happy to move this bit but unfortunately we won't be able to remove the tubes stuff for the moment. Trying to get library by barcode was really messy so I have added material_type to the tube material. If you have a look at GPL-437 in traction service that should give you an idea of what I have done.